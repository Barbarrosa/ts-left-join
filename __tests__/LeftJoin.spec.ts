import LeftJoin from "../src";
import { seed, random } from 'faker';

seed(7437599372);

const leftField = random.word();
const rightField = random.word();
const valueOne = random.word();
const valueTwo = random.word();

test.each([
    [
        'empty left',
        'empty right',
        [],
        [],
    ],
    [
        'one left',
        'empty right',
        [{[leftField]: valueOne}],
        [],
    ],
    [
        'empty left',
        'one right',
        [],
        [{[rightField]: valueOne}],
    ],
    [
        'one left',
        'one right non-match',
        [{[leftField]: valueOne}],
        [{[rightField]: valueTwo}],
    ],
    [
        'one left',
        'one right match',
        [{[leftField]: valueOne}],
        [{[rightField]: valueOne}],
    ],
    [
        'two left',
        'one right match',
        [{[leftField]: valueOne},{[leftField]: valueTwo}],
        [{[rightField]: valueOne}],
    ],
    [
        'two left',
        'two right same match',
        [{[leftField]: valueOne},{[leftField]: valueOne}],
        [{[rightField]: valueOne}],
    ],
    [
        'two left',
        'two right different match',
        [{[leftField]: valueOne},{[leftField]: valueTwo}],
        [{[rightField]: valueTwo},{[rightField]: valueOne}],
    ],
    [
        'one left',
        'two right multiple match',
        [{[leftField]: valueOne}],
        [{[rightField]: valueOne},{[rightField]: valueOne}],
    ],
    [
        'two left',
        'three right different match w/ one multiple',
        [{[leftField]: valueOne},{[leftField]: valueTwo}],
        [{[rightField]: valueTwo},{[rightField]: valueOne},{[rightField]: valueTwo}],
    ],
])('%s, %s',
 <T extends { [key:string]: any}>(
    leftName: string,
    rightName: string,
    leftCollection: T[],
    rightCollection: T[]
 ) => {
    expect(
        LeftJoin(
            leftName,
            leftCollection,
            rightName,
            rightCollection,
            leftField,
            rightField
        )
    ).toMatchSnapshot();
});