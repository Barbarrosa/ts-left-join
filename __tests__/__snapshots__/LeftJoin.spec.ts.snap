// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty left, empty right 1`] = `Array []`;

exports[`empty left, one right 1`] = `Array []`;

exports[`one left, empty right 1`] = `
Array [
  Object {
    "one left": Object {
      "Connecticut": "redundant",
    },
  },
]
`;

exports[`one left, one right match 1`] = `
Array [
  Object {
    "one left": Object {
      "Connecticut": "redundant",
    },
    "one right match": Object {
      "Ferry": "redundant",
    },
  },
]
`;

exports[`one left, one right non-match 1`] = `
Array [
  Object {
    "one left": Object {
      "Connecticut": "redundant",
    },
  },
]
`;

exports[`one left, two right multiple match 1`] = `
Array [
  Object {
    "one left": Object {
      "Connecticut": "redundant",
    },
    "two right multiple match": Object {
      "Ferry": "redundant",
    },
  },
  Object {
    "one left": Object {
      "Connecticut": "redundant",
    },
    "two right multiple match": Object {
      "Ferry": "redundant",
    },
  },
]
`;

exports[`two left, one right match 1`] = `
Array [
  Object {
    "one right match": Object {
      "Ferry": "redundant",
    },
    "two left": Object {
      "Connecticut": "redundant",
    },
  },
  Object {
    "two left": Object {
      "Connecticut": "Oval",
    },
  },
]
`;

exports[`two left, three right different match w/ one multiple 1`] = `
Array [
  Object {
    "three right different match w/ one multiple": Object {
      "Ferry": "redundant",
    },
    "two left": Object {
      "Connecticut": "redundant",
    },
  },
  Object {
    "three right different match w/ one multiple": Object {
      "Ferry": "Oval",
    },
    "two left": Object {
      "Connecticut": "Oval",
    },
  },
  Object {
    "three right different match w/ one multiple": Object {
      "Ferry": "Oval",
    },
    "two left": Object {
      "Connecticut": "Oval",
    },
  },
]
`;

exports[`two left, two right different match 1`] = `
Array [
  Object {
    "two left": Object {
      "Connecticut": "redundant",
    },
    "two right different match": Object {
      "Ferry": "redundant",
    },
  },
  Object {
    "two left": Object {
      "Connecticut": "Oval",
    },
    "two right different match": Object {
      "Ferry": "Oval",
    },
  },
]
`;

exports[`two left, two right same match 1`] = `
Array [
  Object {
    "two left": Object {
      "Connecticut": "redundant",
    },
    "two right same match": Object {
      "Ferry": "redundant",
    },
  },
  Object {
    "two left": Object {
      "Connecticut": "redundant",
    },
    "two right same match": Object {
      "Ferry": "redundant",
    },
  },
]
`;
